rpcserver.go

```diff
@@ -1445,6 +1447,88 @@ func extractOpenChannelMinConfs(in *lnrpc.OpenChannelRequest) (int32, error) {
 	}
 }
 
+// newFundingShimAssembler returns a new fully populated
+// chanfunding.CannedAssembler using a FundingShim obtained from an RPC caller.
+func newFundingShimAssembler(chanPointShim *lnrpc.ChanPointShim,
+	initiator bool, keyRing keychain.KeyRing) (chanfunding.Assembler, error) {
+
+	// Perform some basic sanity checks to ensure that all the expected
+	// fields are populated.
+	switch {
+	case chanPointShim.RemoteKey == nil:
+		return nil, fmt.Errorf("remote key not set")
+
+	case chanPointShim.LocalKey == nil:
+		return nil, fmt.Errorf("local key desc not set")
+
+	case chanPointShim.LocalKey.RawKeyBytes == nil:
+		return nil, fmt.Errorf("local raw key bytes not set")
```

Comment:
If a key locator is provided, then this still fails. The method states that either can be provided.

Reply:
Was left over from where it was copied from (signerrpc), we need both to be populated as is atm, which isn't a big burden as when it's obtained from the wallekitrpc, both are always present.